packages/integration-sdk-runtime/src/logger/index.ts

```diff
@@ -260,14 +271,12 @@ export class IntegrationLogger extends EventEmitter
   }
 
   stepFailure(step: StepMetadata, err: Error) {
-    const name = 'step_failure';
+    const eventName = 'step_failure';
     const { errorId, description } = createErrorEventDescription(
       err,
       `Step "${step.name}" failed to complete due to error.`,
     );
-
-    this.error({ errorId, err }, description);
-    this.publishEvent({ name, description });
+    this.handleFailure({ eventName, errorId, err, description });
```

Comment:
I originally removed the unnecessary `step` log property so that I could create a single consistent interface, sharing logic for both `stepFailure` and `validationFailure`.